pkg/kapp/cmd/tools/inspect_tree_view.go

```diff
@@ -178,14 +178,18 @@ func (a *assocSortingValue) uidOwnersStr() string {
 			foundRes, found := a.rsByUID[string(ref.UID)]
 			if found {
 				// only nest into first object that we find
-				uids = append([]string{string(ref.UID)}, uids...)
+				identifiers = append([]string{a.resIdentifier(foundRes)}, identifiers...)
 				nextRes = &foundRes
 				break
 			}
 		}
 	}
 
-	return "ref-" + strings.Join(uids, "/")
+	return "ref-" + strings.Join(identifiers, "/")
+}
+
+func (a *assocSortingValue) resIdentifier(resource ctlres.Resource) string {
+	return fmt.Sprintf("%s-%s", resource.Kind(), resource.Name())
```

Comment:
```suggestion
	return fmt.Sprintf("%s+%s+%s+%s", resource.Namespace(), resource.APIGroup(), resource.Kind(), resource.Name())
```
Ensures that the identifier is unique and avoids edge cases where the '-' separator might be confused for a dash used in a name.